qir/qat/ValidationPass/ValidationPassConfiguration.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "Commandline/ConfigurationManager.hpp"
+#include "QatTypes/QatTypes.hpp"
+#include "ValidationPass/ValidationPassConfiguration.hpp"
+
+namespace microsoft
+{
+namespace quantum
+{
+
+    void ValidationPassConfiguration::setup(ConfigurationManager& config)
+    {
+        config.setSectionName("Validation configuration", "");
+        config.addParameter(
+            allow_internal_calls_, "allow-internal-calls", "Whether or not internal calls are allowed.");
+        config.addParameter(
+            save_report_to_, "save-validation-report", "Saves the validation report to specified filename.");
+    }
+
+    String ValidationPassConfiguration::profileName() const
+    {
+        return profile_name_;
+    }
+
+    ValidationPassConfiguration ValidationPassConfiguration::fromProfileName(String const& name)
+    {
+
+        auto profile = ValidationPassConfiguration();
+        if (name == "generic")
+        {
+            profile.opcodes_               = Set{};
+            profile.external_calls_        = Set{};
+            profile.allowed_pointer_types_ = Set{};
+
+            profile.allowlist_opcodes_        = true;
+            profile.allowlist_external_calls_ = false;
+            profile.allow_internal_calls_     = true;
+            profile.allowlist_pointer_types_  = false;
+            profile.allow_primitive_return_   = true;
+            profile.allow_struct_return_      = true;
+            profile.allow_pointer_return_     = true;
+        }
+        else if (name == "base")
+        {
+            profile.allow_internal_calls_     = false;
+            profile.allowlist_external_calls_ = true;
+            profile.allowlist_opcodes_        = true;
+            profile.opcodes_        = Set{"br", "call", "unreachable", "ret", "inttoptr"}; //  "phi", "select",
+            profile.external_calls_ = Set{
+                "__quantum__qis__cnot__body",  "__quantum__qis__cz__body", "__quantum__qis__cx__body",
+                "__quantum__qis__cy__body",    "__quantum__qis__h__body",  "__quantum__qis__s__body",
+                "__quantum__qis__s__adj",      "__quantum__qis__t__body",  "__quantum__qis__t__adj",
+                "__quantum__qis__x__body",     "__quantum__qis__y__body",  "__quantum__qis__z__body",
+                "__quantum__qis__rx__body",    "__quantum__qis__ry__body", "__quantum__qis__rz__body",
+                "__quantum__qis__reset__body", "__quantum__qis__mz__body", "__quantum__qis__read_result__body",
+
+            };
+            profile.allowlist_pointer_types_ = true;
+            profile.allowed_pointer_types_   = {"i8*",    "i16*",    "i32*",    "i64*",
```

Comment:
Out of curiousity, what operations end up using `i16*`? I wasn't aware we had any.

Reply:
I am not sure we use it anywhere, but neither do I think we use `i32*` and `i64*`. However, it seems reasonable to allow pointers to all of these primitive types if you allow pointers to one given that we should not base the profile on what Q# currently does, but rather think of it as something that will be used by many different frontends.